neqo-http3/src/connection_server.rs

```diff
@@ -182,11 +270,34 @@ impl Http3ServerHandler {
 
     fn handle_stream_readable(&mut self, conn: &mut Connection, stream_id: u64) -> Res<()> {
         match self.base_handler.handle_stream_readable(conn, stream_id)? {
-            ReceiveOutput::PushStream => Err(Error::HttpStreamCreation),
+            ReceiveOutput::NewStream(NewStreamType::Push(_)) => Err(Error::HttpStreamCreation),
+            ReceiveOutput::NewStream(NewStreamType::Http) => {
+                self.base_handler.add_streams(
+                    stream_id,
+                    Box::new(SendMessage::new(
+                        stream_id,
+                        self.base_handler.qpack_encoder.clone(),
+                        Box::new(self.events.clone()),
+                    )),
+                    Box::new(RecvMessage::new(
+                        MessageType::Request,
+                        stream_id,
+                        Rc::clone(&self.base_handler.qpack_decoder),
+                        Box::new(self.events.clone()),
+                        None,
+                        PriorityHandler::new(false, Priority::default()),
+                        true,
+                    )),
+                );
+                let res = self.base_handler.handle_stream_readable(conn, stream_id)?;
+                assert_eq!(ReceiveOutput::NoOutput, res);
+                Ok(())
+            }
             ReceiveOutput::ControlFrames(control_frames) => {
                 for f in control_frames {
                     match f {
-                        HFrame::MaxPushId { .. } => {
+                        HFrame::MaxPushId { .. } | HFrame::Settings { .. } => {
+                            // Settings frame is only handled by the client.
```

Comment:
This is a worrying comment.  I think that you are saying that on the client needs to have anything other than the base handler process a SETTINGS frame; or that the base handler logic is sufficient for a server.